javaslang/src-gen/main/java/javaslang/Tuple.java

```diff
@@ -223,4 +207,73 @@ static Tuple0 empty() {
     static <T1, T2, T3, T4, T5, T6, T7, T8> Tuple8<T1, T2, T3, T4, T5, T6, T7, T8> of(T1 t1, T2 t2, T3 t3, T4 t4, T5 t5, T6 t6, T7 t7, T8 t8) {
         return new Tuple8<>(t1, t2, t3, t4, t5, t6, t7, t8);
     }
+
+    static <T1> Tuple1<Seq<T1>> sequence1(Iterable<? extends Tuple1<? extends T1>> tuples) {
+        Objects.requireNonNull(tuples, "tuples is null");
+        return new Tuple1<>(Stream.ofAll(tuples).map(Tuple1::_1));
+    }
+
+    static <T1, T2> Tuple2<Seq<T1>, Seq<T2>> sequence2(Iterable<? extends Tuple2<? extends T1, ? extends T2>> tuples) {
+        Objects.requireNonNull(tuples, "tuples is null");
+        return new Tuple2<>(Stream.ofAll(tuples).map(Tuple2::_1),
+                            Stream.ofAll(tuples).map(Tuple2::_2));
+    }
+
+    static <T1, T2, T3> Tuple3<Seq<T1>, Seq<T2>, Seq<T3>> sequence3(Iterable<? extends Tuple3<? extends T1, ? extends T2, ? extends T3>> tuples) {
+        Objects.requireNonNull(tuples, "tuples is null");
+        return new Tuple3<>(Stream.ofAll(tuples).map(Tuple3::_1),
+                            Stream.ofAll(tuples).map(Tuple3::_2),
+                            Stream.ofAll(tuples).map(Tuple3::_3));
```

Comment:
Since the args length is decreased now
```java
return new Tuple3<>(s.map(Tuple3::_1), s.map(Tuple3::_2), s.map(Tuple3::_3));
```
Can we have the entire statement in one line?